packages/orama/src/components/facets.ts

```diff
@@ -56,23 +57,64 @@ export async function getFacets(
     for (const facet of facetKeys) {
       const facetValue = facet.includes('.')
         ? (await getNested<string>(doc!, facet))!
-        : (doc![facet] as number | boolean)
-
-      // Range facets based on numbers
-      if (properties[facet] === 'number') {
-        for (const range of (facetsConfig[facet] as NumberFacetDefinition).ranges) {
-          if ((facetValue as number) >= range.from && (facetValue as number) <= range.to) {
-            if (facets[facet].values[`${range.from}-${range.to}`] === undefined) {
-              facets[facet].values[`${range.from}-${range.to}`] = 1
-            } else {
-              facets[facet].values[`${range.from}-${range.to}`]++
+        : (doc![facet] as SearchableValue)
+
+      const alreadyInsertedValues = new Set()
+      switch (properties[facet]) {
+        case 'number': {
+          for (const range of (facetsConfig[facet] as NumberFacetDefinition).ranges) {
+            if ((facetValue as number) >= range.from && (facetValue as number) <= range.to) {
+              if (facets[facet].values[`${range.from}-${range.to}`] === undefined) {
+                facets[facet].values[`${range.from}-${range.to}`] = 1
+              } else {
+                facets[facet].values[`${range.from}-${range.to}`]++
+              }
+            }
+          }
+          break
+        }
+        case 'boolean':
+        case 'string': {
+          // String or boolean based facets
+          const value = facetValue?.toString() ?? (properties[facet] === 'boolean' ? 'false' : '')
+          facets[facet].values[value] = (facets[facet].values[value] ?? 0) + 1
+          break
+        }
+        case 'boolean[]':
+        case 'string[]': {
+          for (const v of (facetValue as Array<string>)) {
+            // String or boolean based facets
+            const value = v?.toString() ?? (properties[facet] === 'boolean' ? 'false' : '')
+            if (alreadyInsertedValues.has(value)) {
+              continue
+            }
+
+            facets[facet].values[value] = (facets[facet].values[value] ?? 0) + 1
+
+            alreadyInsertedValues.add(value)
+          }
+          break
+        }
+        case 'number[]': {
```

Comment:
Nit: for easier readability and maintainability, can you please move the `number` case right above this so they are "grouped"?

Reply:
I think @allevo put this here as he's grouping lists after the primitives, isn't that ok?

Reply:
Done, after the rework, the number and number[] cases are very similar. It makes sense to put them together.